### PR TITLE
refactor: allow 18 decimals in the EVM

### PR DESCRIPTION
**Pull request by jorgemmsilva**

### CODE REVIEW

**Review by lmoe — APPROVED** [review]

File: packages/solo/run.go [hunk 1]
```diff
@@ -68,7 +68,6 @@ func (ch *Chain) runTaskNoLock(reqs []isc.Request, estimateGas bool) *vm.VMTaskR
 		// state baseline is always valid in Solo
 		EnableGasBurnLogging: ch.Env.enableGasBurnLogging,
 		EstimateGasMode:      estimateGas,
-		MigrationsOverride:   ch.migrationScheme,
```
Comment (dessaya) [inline packages/solo/run.go:71]: Why this change?

File: packages/util/misc.go [hunk 2]
```diff
@@ -34,6 +34,10 @@ func IsZeroBigInt(bi *big.Int) bool {
 	return len(bi.Bits()) == 0
 }
 
+func IsPositiveBigInt(n *big.Int) bool {
+	return n.Cmp(Big0) == 1
```
Comment (dessaya) [inline packages/util/misc.go]: `n.Sign() > 0`?

File: packages/vm/core/accounts/fungibletokens.go [hunk 3]
```diff
@@ -47,6 +47,23 @@ func creditToAccount(state kv.KVStore, accountKey kv.Key, assets *isc.Assets) {
 	}
 }
 
+// CreditToAccount brings new funds to the on chain ledger
+// NOTE: this function does not take NFTs into account
```
Comment (dessaya) [inline packages/vm/core/accounts/fungibletokens.go]: The `NOTE` is probably not needed since the parameter is already `big.Int`, not `Assets`

File: packages/solo/solotest/snapshot.db
Comment (dessaya) [inline packages/solo/solotest/snapshot.db:1]: Is this intended to be committed?

**Review by dessaya — APPROVED** [review]
So this change allows accounts to own fractions of a glow, but still on L1 the ledger must be balanced, right? I mean, it must still hold that the sum of all L2 accounts is equal to the L1 assets owned by the chain.

IIRC there are functions in `solo` and `vmimpl` that assert that invariant. Maybe those need to be adapted to include the extra decimals in the sum?

See `CheckLedger` in `packages/vm/core/accounts/checkledger.go` and `MustBalanced` in `packages/vm/vmtxbuilder/totals.go`

File: packages/solo/run.go (hunk 1 above)
Comment (jorgemmsilva) [reply packages/solo/run.go:71]: it's not much of an "override" if it always has a value, right? :stuck_out_tongue: 
it was causing issues when testing the migration

File: packages/solo/solotest/snapshot.db
Comment (jorgemmsilva) [reply packages/solo/solotest/snapshot.db:1]: no, thank you for pointing that out :facepalm:

**Comment by jorgemmsilva** [comment]
@dessaya thanks for the review, good points.
`packages/vm/vmtxbuilder/totals.go` shouldn't need to be updated, as it only uses information from the transaction (inputs/outputs)

`packages/vm/core/accounts/checkledger.go` does make sense to update, working on it :+1:
